lib/rules/unicode-bom/__tests__/index.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+const rule = require("..");
+const { messages, ruleName } = rule;
+
+testRule(rule, {
+  ruleName,
+  config: ["always"],
+  skipBasicChecks: true,
+
+  accept: [
+    {
+      code: "\uFEFF",
+      description: "empty with BOM"
+    },
+    {
+      code: "\uFEFFa{}",
+      description: "with BOM"
+    }
+  ],
+
+  reject: [
+    {
+      code: "",
+      description: "empty without BOM",
+      message: messages.expected
+    },
+    {
+      code: "a{}",
+      description: "without BOM",
+      message: messages.expected
+    }
+  ]
+});
+
+testRule(rule, {
+  ruleName,
+  config: ["never"],
+  skipBasicChecks: true,
+
+  accept: [
+    {
+      code: "",
+      description: "empty without BOM"
+    },
+    {
+      code: "a{}",
+      description: "without BOM"
+    }
+  ],
+
+  reject: [
+    {
+      code: "\uFEFF",
+      description: "empty with BOM",
+      message: messages.rejected
+    },
+    {
+      code: "\uFEFFa{}",
+      description: "with BOM",
+      message: messages.rejected
+    }
+  ]
+});
```

Comment:
let's add couple test with `html` and `css-in-js`

Reply:
Done.

Reply:
Done.